### PR TITLE
xdg-shell stable

### PR DESCRIPTION
**Pull request by keithbowes**
Updates to use xdg-shell stable.

### CODE REVIEW

**Comment by majestrate** [comment]
which compositors support this so far?

**Comment by keithbowes** [comment]
The hard part is to find compositors that don't support it :)  Plus, it's pretty much the same as xdg-shell-unstable-v6, with only the type, interface, and function names changed (the file generated by wayland-scanner maps them to the exact same Wayland API functions).  I downloaded an old version of Weston that supported v6 only and confirmed that it works fine in it too.

BTW, the Sway developers want to get rid of support for v6.  An incomplete list of compositors/compositor libraries and xdg-shell support:
* Weston - v6 and stable (6.0+); v6 only (4.0/5.0); v5 and v6 (1.12 to 3.0); v5 only (<1.12)
* Sway - v6 and stable (1.0+; older versions used wlc)
* wlc (defunct, with compositors written in it being rewritten in wlroots or smithay) - v6 only (0.0.7+); v5 only (<0.0.7)
* swc (used primarily by Velox) - stable only (since May 26, 2018); v5 only (before that)
* smithay - v6 and stable

In summary, it should work in anything from November 2016 onward (the exception being swc, which took until late May 2018).  v5 was taken out of everything ages ago.  I did remove the v5 code entirely, but I suppose it could be put back under a different name, if there's some ideological principle in supporting ancient compositors.  I personally think it's more important for it to work in newer ones than in older ones; after all, part of the ideology behind Wayland is not to support old cruft forever (like X does).

EDIT:  Merging this should solve issue #3 and issue #7 (for a while now, swc has only supported xdg-shell stable).

**Comment by majestrate** [comment]
On Wed, Jun 19, 2019 at 12:59:38PM -0700, Keith Bowes wrote:
> The hard part is to find compositors that don't support it :)  Plus, it's pretty much the same as xdg-shell-unstable-v6, with only the type, interface, and function names changed (the file generated by wayland-scanner maps them to the exact same Wayland API functions).  I downloaded an old version of Weston that supported v6 only and confirmed that it works fine in it too.
> 
> BTW, the Sway developers want to get rid of support for v6.  An incomplete list of compositors/compositor libraries and xdg-shell support:
> * Weston - v6 and stable (6.0+); v6 only (4.0/5.0); v5 and v6 (1.12 to 3.0); v5 only (<1.12)
> * Sway - v6 and stable (1.0+; older versions used wlc)
> * wlc (defunct, with compositors written in it being rewritten in wlroots or smithay) - v6 only (0.0.7+); v5 only (<0.0.7)
> * swc (used primarily by Velox) - stable only (since May 26, 2018); v5 only (before that)
> * smithay - v6 and stable
> 
> In summary, it should work in anything from November 2016 onward (the exception being swc, which took until late May 2018).  v5 was taken out of everything ages ago.  I did remove the v5 code entirely, but I suppose it could be put back under a different name, if there's some ideological principle in supporting ancient compositors.  I personally think it's more important for it to work in newer ones than in older ones; after all, part of the ideology behind Wayland is not to support old cruft forever (like X does).
gotcha
